int16 naive_time
int16 new_jumptable_time